atom/browser/net/atom_url_request.cc

```diff
@@ -504,4 +504,17 @@ void AtomURLRequest::InformDelegateErrorOccured(const std::string& error,
     delegate_->OnError(error, isRequestError);
 }
 
+void AtomURLRequest::GetUploadProgress(mate::Dictionary& progress) {
+  if (!request_) {
+    progress.Set("started", false);
+    progress.Set("current", 0);
+    progress.Set("total", 0);
+  } else {
+    progress.Set("started", true);
+    auto upload_progress = request_->GetUploadProgress();
+    progress.Set("current", upload_progress.position());
+    progress.Set("total", upload_progress.size());
+  }
```

Comment:
since `mate::Dictionary::Set()` is a template method, it looks like here we're sometimes setting current/total to `int` and sometimes to `uint64_t`. At first read I think that will make the former invoke `Converter::ToV8(isolate,int32_t)` and the latter `Converter::ToV8(uint64_t)`. Does that matter?

**If** that matters, we could consistently use net::UploadProgress' types:

```cpp
net::UploadProgress upload_progress;
if (request_) {
  progress.set("started", true);
  upload_progress = request_->GetUploadProgress();
} else {
  progress.set("started", false);
}
progress.set("current", upload_progress.position());
progress.set("total", upload_progress.size());
```

Reply:
Not sure it matters, but the consistent code looks nicer anyway 👍